src/dkg/round1.rs

```diff
@@ -121,11 +129,114 @@ pub fn import_secret_package(
     SerializableSecretPackage::deserialize_from(&serialized[..]).map(|pkg| pkg.into())
 }
 
+#[must_use]
+fn input_checksum<I>(min_signers: u16, signing_participants: &[I]) -> Checksum
+where
+    I: Borrow<Identity>,
+{
+    let mut hasher = SipHasher24::new();
+
+    hasher.write(&min_signers.to_le_bytes());
+
+    let mut signing_participants = signing_participants
+        .iter()
+        .map(Borrow::borrow)
+        .collect::<Vec<_>>();
+    signing_participants.sort_unstable();
+    signing_participants.dedup();
+
+    for id in signing_participants {
+        hasher.write(&id.serialize());
+    }
+
+    hasher.finish()
+}
+
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct Package {
```

Comment:
Nit: can we call this `PublicPackage` or similar? I know that frost-core calls it Package, but I find their terminology quite annoying, given that there are multiple "packages" produced in round 1.

Reply:
That sounds good to me, and it avoids collision with frost-core nomenclature